tests/unit_tests/test_suassystem_sda_converter.py

```diff
@@ -7,31 +7,129 @@
 class SDAConverterTestCase(unittest.TestCase):
 
     def setUp(self):
-
-        #Initialize
-
         self.setup_client_converter()
 
     def setup_client_converter(self):
-        self.initial_coordinates = Location(38.8703041, -77.3214035, 60.950000762939453)
+        self.VectorMath = VectorMath()
+        self.initial_coordinates = Location(38.8703041, -77.3214035, 0)
         self.boundary_points = [
             {
                 "boundary_pts" : [
-                    {"latitude" : 38.867580, "longitude" : -77.330360, "order" : 0},
-                    {"latitude" : 38.876535, "longitude" : -77.330060, "order" : 0},
-                    {"latitude" : 38.877002, "longitude" : -77.314997, "order" : 0},
-                    {"latitude" : 38.867513, "longitude" : -77.315769, "order" : 0}
+                    {"latitude" : 38.8697558739, "longitude" : -77.3221076688, "order" : 0},
+                    {"latitude" : 38.8708523261, "longitude" : -77.3221076743, "order" : 1},
+                    {"latitude" : 38.8697558739, "longitude" : -77.3206993312, "order" : 2},
+                    {"latitude" : 38.8708523261, "longitude" : -77.3206993257, "order" : 3}
                 ]
             }
         ]
 
         self.sda_converter = SDAConverter(self.initial_coordinates, self.boundary_points)
 
+    def test_convert_fly_zones(self):
+        converted_fly_zones = self.sda_converter.convert_fly_zones(self.boundary_points)
+        for num in range(len(converted_fly_zones)):
+            for index in range(len(converted_fly_zones[num])):
+                print(self.boundary_points[num]["boundary_pts"][index]["latitude"], self)
+                location = Location(self.boundary_points[num]["boundary_pts"][index]["latitude"], self.boundary_points[num]["boundary_pts"][index]["longitude"], 0)
+                expected_converted_boundary_point = haversine(self.initial_coordinates, location, units="US")
+
+                self.assertEqual(converted_fly_zones[num][index][0], expected_converted_boundary_point[0])
+                self.assertEqual(converted_fly_zones[num][index][1], expected_converted_boundary_point[1])
+    
+    def test_avoid_obstacles(self):
+        self.sda_converter.obstacle_map.set_drone_position(numpy.array([0,0,0]))
+        self.sda_converter.reset_obstacles()
+        self.sda_converter.obstacle_map.add_obstacle(StationaryObstacle(numpy.array([40,0,0]),5,100))
+        test_waypoint = Location(38.8703041, -77.3210866228, 0)
+        self.sda_converter.set_waypoint(test_waypoint)
+        self.sda_converter.avoid_obstacles()
+        boolean, possible_paths = self.sda_converter.obstacle_map.is_obstacle_in_path()
```

Comment:
You should also be testing the boolean value to ensure you get the correct behavior